tests/analysis/test_transport_mode_identification.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+import os
+
+import pytest
+import numpy as np
+import geopandas as gpd
+from numpy.testing import assert_almost_equal
+
+import trackintel as ti
+from trackintel.io.dataset_reader import read_geolife
+
+
+class TestTransportModeIdentification:
+    def test_check_empty_dataframe(self):
+        """Assert that the method does not work for empty DataFrames (but
+        that the rest works fine, e.g., method signature)."""
+        tpls = ti.read_triplegs_csv('tests/data/triplegs_transport_mode_identification.csv', sep=';')
+        empty_frame = tpls[0:0]
+        with pytest.raises(AssertionError):
+            empty_frame.as_triplegs.predict_transport_mode(method='simple-coarse')
+   
+
+    def test_check_correct_simple_coarse_identification(self):
+        """Assert that the simple-coarse transport mode identification yields the correct results."""
```

Comment:
I think it would be better to split this into different tests

Reply:
Ok so one test per crs case?

Reply:
One per crs case could be a good idea. 
A bit hard to say how to split the function as it is a bit flexible. The idea is to have 1 function per functionality that you want to test. E.g., you could have one case to test correct warning behavior, one to test correct classification behavior (for different crs cases) or one per crs. A good rule of thumb is that you should be able to explain the test in a single short sentence (this should then be used to document the test). 
E.g., "Testing correct warning behavior" or "Testing correct behavior for crs=4326"

Reply:
see last commit